specification/app/resource-manager/Microsoft.App/stable/2025-07-01/ManagedEnvironments.json

```diff
@@ -0,0 +1,1754 @@
+              ]
+            },
+            "customDomainConfiguration": {
+              "$ref": "./CommonDefinitions.json#/definitions/CustomDomainConfiguration",
+              "description": "Custom domain configuration for the environment"
+            },
+            "eventStreamEndpoint": {
+              "description": "The endpoint of the eventstream of the Environment.",
+              "type": "string",
+              "readOnly": true
+            },
+            "workloadProfiles": {
+              "description": "Workload profiles configured for the Managed Environment.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/WorkloadProfile"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "kedaConfiguration": {
+              "$ref": "#/definitions/KedaConfiguration",
+              "description": "The configuration of Keda component."
+            },
+            "daprConfiguration": {
+              "$ref": "#/definitions/DaprConfiguration",
+              "description": "The configuration of Dapr component."
+            },
+            "infrastructureResourceGroup": {
+              "description": "Name of the platform-managed resource group created for the Managed Environment to host infrastructure resources. If a subnet ID is provided, this resource group will be created in the same subscription as the subnet.",
+              "type": "string",
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            },
+            "peerAuthentication": {
+              "description": "Peer authentication settings for the Managed Environment",
+              "type": "object",
+              "properties": {
+                "mtls": {
+                  "description": "Mutual TLS authentication settings for the Managed Environment",
+                  "$ref": "#/definitions/Mtls"
+                }
+              }
+            },
+            "peerTrafficConfiguration": {
+              "description": "Peer traffic settings for the Managed Environment",
+              "type": "object",
+              "properties": {
+                "encryption": {
+                  "description": "Peer traffic encryption settings for the Managed Environment",
+                  "type": "object",
+                  "properties": {
+                    "enabled": {
+                      "type": "boolean",
+                      "description": "Boolean indicating whether the peer traffic encryption is enabled"
+                    }
+                  }
+                }
+              }
+            },
+            "privateEndpointConnections": {
+              "description": "Private endpoint connections to the resource.",
+              "type": "array",
+              "items": {
+                "$ref": "./CommonDefinitions.json#/definitions/PrivateEndpointConnection"
```

Comment:
>./CommonDefinitions.json#/definitions/PrivateEndpointConnection" [](http://example.com/codeflow?start=25&length=64)

Use v5/v6 common types?

Reply:
We do not want to use common types for private endpoint connections.  The PE related spec is added from https://github.com/Azure/azure-rest-api-specs/pull/28775, where we already have a discussion for using common type or not. In previous discussion, we do not want to use private link common types because we have our own values and description for some properties. For example, we should have our own definition for PrivateEndpointConnectionProvisioningState which should align with our server side. And there are many RPs using self-defined private endpoint connection for flexiblity.

For now, we do need to be consistent with previous api version because the feature is already public available and we do not want to break existing customers.